dvc/utils/__init__.py

```diff
@@ -480,3 +481,48 @@ def glob_targets(targets, glob=True, recursive=True):
         for target in targets
         for exp_target in iglob(target, recursive=recursive)
     ]
+
+
+@contextmanager
+def intercept_error(onerror: Callable = None, **kwargs):
+    try:
+        yield
+    except Exception as e:  # pylint: disable=W0703
+        logger.debug("", exc_info=True)
+        if onerror is not None:
+            onerror(e, **kwargs)
+
+
+class Onerror:
+    def __init__(self):
+        self.errors = {}
+
+    def __call__(self, exception: Exception, revision: str, path: str = None):
```

Comment:
Also, this is just my personal preference, but I'm not a big fan of using `__call__` in this way since I think it makes it less obvious what this is (a handler for the `onerror` callback method). 

I get that this:

```
class Onerror():
    def __call__(self, ...):
        ...

onerror = Onerror()
Repo.func(onerror=onerror)
```

is slightly less verbose than

```
class ErrorCollector():
    def onerror(self, ...):
        ...

errors = ErrorCollector()
Repo.func(onerror=errors.onerror)
```

but I think the 2nd way is more obvious about what it is doing